clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10134,6 +10137,24 @@ def note_format_fix_specifier : Note<"did you mean to use '%0'?">;
 def note_printf_c_str: Note<"did you mean to call the %0 method?">;
 def note_format_security_fixit: Note<
   "treat the string as an argument to avoid this">;
+def warn_format_cmp_role_mismatch : Warning<
```

Comment:
This diagnostic has a lot of combinations but it looks like you only test two of them. We should cover all of them.